dstu3/src/main/java/org/opencds/cqf/dstu3/providers/ActivityDefinitionApplyProvider.java

```diff
@@ -147,7 +147,11 @@ private ProcedureRequest resolveProcedureRequest(ActivityDefinition activityDefi
         ProcedureRequest procedureRequest = new ProcedureRequest();
         procedureRequest.setStatus(ProcedureRequest.ProcedureRequestStatus.DRAFT);
         procedureRequest.setIntent(ProcedureRequest.ProcedureRequestIntent.ORDER);
-        procedureRequest.setSubject(new Reference(patientId));
+        String patientReferenceString = patientId;
+        if (!patientId.startsWith("Patient/")) {
+            patientReferenceString = "Patient/" + patientId;
```

Comment:
Reasonable, but what about if patientId is a contained reference? (e.g. `#1234`). And it should probably also account for the possibility that the reference is an absolute URL (e.g. `http://blah-be-de-blah...` and `uuid:oid:XXX...`)

Reply:
Used URI class to account for absoluteUrl and references to contained resources (indicated by presence of a fragment).

Reply:
Okay, but it does still need to check for the case that the input is relative, but already starts with `Patient/`

Reply:
Restored that check and also updated R4 with these same changes.

Reply:
Looks good